### PR TITLE
Refactor ConfigModel Construction in StandardActivity

### PR DESCRIPTION
**Pull request by dave2wave**
This will have a refactoring of StandardActivity to provide proper scoping of ConfigModel construction along with providing advice and analysis about parameters.

We will try to solve all of these in this PR:

- [x] #1808 
- [x] #1609
- [x] #1187
- [x] #1182
- [ ] #1003
- [ ] #451
- [ ] #117
- [x] #21

### CODE REVIEW

**Comment by dave2wave** [comment]
--advisor=validate or --parameter-model will be used to show activity, template, and model parameters use.

File: nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ActivityDef.java [hunk 1]
```diff
@@ -230,4 +242,65 @@ public NBLabels auxLabels() {
...
 
     }
+
+    public NBConfigModel getConfigModel() {
+        ConfigModel cfgmodel = ConfigModel.of(this.getClass());
+        Map<String, String> params = parameterMap.getStringStringMap();
+        params.forEach((k, v) -> {
+            cfgmodel.add(Param.defaultTo(k, v, "activity parameter found on command line"));
+        });
+        if ( ! params.containsKey(FIELD_ALIAS)) {
```
Comment (jshook) [inline nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ActivityDef.java:252]: I'm not entirely sure what this section should be doing with all the conditionals. This is what I'd expect: 1) individual config models are declarative and definitive about something specific, so no conditions are needed on a per-param basis. 2) config models can be derived by disjunction or intersection with others. 3) defaults in the config model are manifested when the config model is asked for the parameter that supports it.

But I might be missing something here. Can you elaborate a bit on what the purpose of this section is?

**Review by jshook — COMMENTED** [review]
There are some nuances here that I would like to understand better.

File: nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ActivityDef.java (hunk 1 above)
Comment (dave2wave) [reply nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ActivityDef.java:252]: `ConfigModel.add` adds keys to a model regardless of whether or not the value has already been added to the model. If the field was already by the `params.forEach` loop then it is skipped. A better fix might be to change `ConfigModel.add` behavior.

The usefulness of having a ConfigModel for ActivityDef is still speculative, but will be more for checking error conditions with excess.

Comment (dave2wave) [reply nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ActivityDef.java]: I've adjusted the behavior in `ConfigModel.add`

**Review by jshook — DISMISSED** [review]
LGTM, Thanks for digging into this!

**Review by MarkWolters — APPROVED** [review]
rubber stamping

**Pull request merged**